src/models/api_key/request.rs

```diff
@@ -1,8 +1,8 @@
+use crate::models::PermissionGrant;
 use serde::{Deserialize, Serialize};
 
 #[derive(Serialize, Deserialize)]
 pub struct ApiKeyRequest {
```

Comment:
I think we need to narrow down api key cration logic to prevent use cases like:
- user with scoped api key permission + api key create permission can create an api key with higher permissions than it has and take over system.

I guess we can do something like: prevent api key creation that has higher permissions than api key used for creating a new api key.